### ISSUE TITLE
Display interactive docs interface for APIs when possible 

### ISSUE THREAD

**Issue by philipashlock** [GSA/datagov-wptheme#147]
When metadata is available to present interactive docs for an API (such as Swagger, RAML, API Blueprint) the dataset page should display this interface. 

**Note:** this depends on new fields in v1.1 of the Project Open Data schema `describedBy` and `describedByType`. See https://github.com/project-open-data/project-open-data.github.io/issues/291#issuecomment-55680064

**Comment by philipashlock** [comment]
@gbinal This comment was imported from feedback we got in testing from last year and it deals with the inconsistencies in API documentation across agencies. Could you provide an update on the work you've done with improving API documentation and encouraging agencies to develop API standards? 

I think a lot of this can also be addressed as part of the updates to the data.json schema, particularly https://github.com/project-open-data/project-open-data.github.io/issues/291#issuecomment-55680064

**Comment by rebeccawilliams** [comment]
Pinging this for the @gbinal / @GUI API standards update so we can port this Issue over and/or close it here.

**Comment by GUI** [comment]
I think Gray might be best suited to answer this. But my quick take: A long time ago, we envisioned that being one thing api.data.gov could offer: a unified API documentation experience (for example, if everyone used Swagger, then this would be pretty trivial). However, as we've seen things grow, I think we've realized it might be more appropriate for each agency to tackle their own documentation as they best see fit outside of api.data.gov's doc pages. This gives them a little more flexibility to tailor their entire developer experience and build out more developer content besides just the straight API docs. This direction also naturally means that the API docs are a little less standardized.

So that doesn't precisely help answer this issue, but I know Gray has also been encouraging agencies to start with some [developer hub templates](http://18f.github.io/API-All-the-X/pages/developer_hub_templates) that are pretty standard, so he might have other thoughts on this front.

**Comment by philipashlock** [comment]
Since we now support the ability to reference machine readable API documentation (like Swagger) from within the data.json metadata (in v1.1), I'm going to change this issue to be about providing the front-end display for that. I think it makes sense for agencies to put a lot of care into crafting good custom documentation on their own site, but if they want to provide something basic and don't already have a good place to present that, it seems like the dataset page on data.gov would be a helpful place to provide the interactive docs interface provided by something like Swagger.

**Comment by gbinal** [comment]
To follow up @gui and to the original, we've got some information that touches on [interactive (especially opensourced) documentation ](http://18f.github.io/API-All-the-X/pages/open_source_documentation), but it doesn't appear that we're in a good place to require or even strongly push standardization of documentation.  The federal API scene is just to disjointed - or rather, is operating so independently.  

That said, yes, what @philipashlock brought up - the idea of enabling an agency's machine-readable API metadata to power interactive documentation on data.gov: that's a great idea.

**Comment by philipashlock** [comment]
@gbinal I agree that we're still far off from requiring this or picking a winner as far as a standard, but we do now provide more detailed guidance about how to specify this documentation using the emerging standards: https://project-open-data.cio.gov/v1.1/api/

Data.gov should also support searching and filtering for APIs that specify machine readable documentation like so we can track adoption and praise early adopters.
